### PR TITLE
Add registry server flag on upgrade

### PR DESCRIPTION
**Pull request by JoeMcGuire98**
### Proposed changes
Currently the upgrade to version (image tag) is baked into the CLI, and the registry server value defaults to whatever the mesh was last deployed with. This change will allow users to override those values with flags in the `upgrade` command.

### Checklist
Before creating a PR, run through this checklist and mark each as complete.

- [x] I have read the [CONTRIBUTING](https://github.com/nginxinc/nginx-service-mesh/blob/main/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that all unit tests pass after adding my changes
- [x] I have updated necessary documentation
- [x] I have rebased my branch onto main
- [x] I will ensure my PR is targeting the main branch and pulling from my branch from my own fork

### CODE REVIEW

File: internal/nginx-meshctl/commands/upgrade.go [hunk 1]
```diff
@@ -36,7 +37,17 @@ var upgradeTimeout = 5 * time.Minute
...
+	var (
+		tagOverride    string
+		serverOverride string
+		dryRun         bool
+	)
+
+	files, _, err := helm.GetBufferedFilesAndValues()
+	if err != nil {
+		log.Fatal(err)
+	}
```
Comment (sjberman) [inline internal/nginx-meshctl/commands/upgrade.go:49]: Why move this outside of the constructor?

**Review by sjberman — COMMENTED** [review]
Be sure to update PR description please.

File: internal/nginx-meshctl/commands/upgrade_test.go [hunk 2]
```diff
@@ -66,15 +68,16 @@ var _ = Describe("Upgrade", func() {
...
 		var err error
 		fakeK8s = fake.NewFakeK8s(v1.NamespaceDefault, shouldSkipRelease)
-		upg, err = newUpgrader(fakeK8s, true)
+		files, values, err := helm.GetBufferedFilesAndValues()
 		Expect(err).ToNot(HaveOccurred())
+		upg = newUpgrader(files, values, fakeK8s, true)
 	})
 
 	Context("upgrades the mesh", func() {
 		It("upgrades the mesh using telemetry config", func() {
```
[the remark this answers is not in the document]
Comment (JoeMcGuire98) [reply internal/nginx-meshctl/commands/upgrade_test.go:75]: I added one for checking the registry field, but I think the `upgrades the mesh using telemetry config` already checks that the version comes out as expected

**Review by chase-kiefer — APPROVED** [review]

File: docs/content/guides/upgrade.md [hunk 3]
```diff
@@ -42,6 +42,10 @@ nginx-meshctl upgrade
 
 This will upgrade the NGINX Service Mesh control plane to the latest version. All user configuration (traffic policies, mesh configuration, deploy configuration) is preserved through the upgrade. New applications will not be injected during the upgrade and existing applications will not receive configuration updates. Existing applications may not function properly until updated. It is recommended that upgrades are only performed during a maintenance window due to the brief downtime.
 
+By default, the mesh will pull images from the registry the mesh was originally deployed with. If you would like to pull from a different registry, you can specify it with the `--registry-server` flag on the CLI.
```
Comment (sjberman) [inline docs/content/guides/upgrade.md:45]:
```suggestion
By default, the mesh will pull images from the registry that it was originally deployed with. If you would like to pull from a different registry, you can use the`--registry-server` flag.
```

File: docs/content/guides/upgrade.md [hunk 4]
```diff
@@ -42,6 +42,10 @@ nginx-meshctl upgrade
...
+
+Additionally, if you would like to upgrade to a custom image tag you can use the `--image-tag` flag with the CLI.
```
Comment (sjberman) [inline docs/content/guides/upgrade.md:47]:
```suggestion
Additionally, if you would like to upgrade to a custom image tag you can use the `--image-tag` flag.
```

File: internal/nginx-meshctl/commands/upgrade.go [hunk 5]
```diff
@@ -233,8 +252,9 @@ func (u *upgrader) upgrade(version string) error {
 // - copy on top of the new release's deploy-time configuration
 // - get previous release's run-time configuration (mesh-config ConfigMap)
 // - copy on top of the new release's deploy-time configuration
-// - set new version.
-func (u *upgrader) buildValues(ctx context.Context, version string) (map[string]interface{}, error) {
+// - set new version
+// - set new registry server if needed.
+func (u *upgrader) buildValues(ctx context.Context, version string, registry string) (map[string]interface{}, error) {
```
Comment (sjberman) [inline internal/nginx-meshctl/commands/upgrade.go:257]: nit: since version and registry are the same type, you only have to specify `string` once.

**Review by sjberman — APPROVED** [review]

**Pull request merged**